core-compose/src/main/java/com/squareup/workflow/ui/compose/WorkflowContainer.kt

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@file:Suppress("EXPERIMENTAL_API_USAGE", "RemoveEmptyParenthesesFromAnnotationEntry")
+
+package com.squareup.workflow.ui.compose
+
+import androidx.compose.Composable
+import androidx.compose.Direct
+import androidx.compose.Pivotal
+import androidx.compose.onDispose
+import androidx.compose.remember
+import androidx.compose.state
+import androidx.ui.core.CoroutineContextAmbient
+import androidx.ui.core.Modifier
+import androidx.ui.foundation.Box
+import androidx.ui.savedinstancestate.Saver
+import androidx.ui.savedinstancestate.SaverScope
+import androidx.ui.savedinstancestate.savedInstanceState
+import com.squareup.workflow.Snapshot
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.compose.ComposeRendering
+import com.squareup.workflow.diagnostic.WorkflowDiagnosticListener
+import com.squareup.workflow.launchWorkflowIn
+import com.squareup.workflow.ui.ViewEnvironment
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.CONFLATED
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import okio.ByteString
+import kotlin.coroutines.CoroutineContext
+
+// TODO this is probably WAY too many overloads
```

Comment:
@rjrjr WDYT?

Reply:
Also, the number of overloads here (8) makes the code completion for this function almost unusable. Intellij doesn't filter out the extra ones, and when presented with 8 giant signatures it's almost impossible to actually tell which variant you're calling and what arguments you're supposed to pass. So there's a trade-off between discoverability/first time UX on one hand and minimal code at call sites on the other.

Reply:
That said, the core library has lots of overloads for stuff like this, so and I don't think we've had negative feedback about them, so maybe I'm overthinking it.